### PR TITLE
fix: add metrics to fetchMeetingInfo and fix tests

### PR DESCRIPTION
**Pull request by rajeshtezu90**
# COMPLETES #<[SPARK-353036](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-353036)>

## This pull request addresses

Adds missing metrics and fixes unit tests except BNR tests

## by making the following changes

All the unit tests for plugin-meetings passes except the BNR tests which is fixed in another PR #2396

### Screenshot
Screenshot 2022-08-11 at 8 37 53 PM

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.js [hunk 1]
```diff
@@ -164,7 +164,7 @@ export default class MeetingInfoV2 {
             stack: err.stack
           }
         );
-        throw new MeetingInfoV2AdhocMeetingError(err.body?.code, err.body?.message);
+        throw err;
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.js:167]: I think we need this back

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 2]
```diff
@@ -968,6 +968,10 @@ export default class Meeting extends StatelessWebexPlugin {
         EVENT_TRIGGERS.MEETING_INFO_AVAILABLE
       );
 
+      Metrics.sendBehavioralMetric(
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:971]: Can you run this with @Neeraj-swarnkar 

i dont think its the right place

**Review by arun3528 — CHANGES_REQUESTED** [review]

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 3]
```diff
@@ -1023,13 +1035,12 @@ export default class Meeting extends StatelessWebexPlugin {
...
       password, captchaCode
     })
-      .then(() => {
-        Metrics.sendBehavioralMetric(
-          BEHAVIORAL_METRICS.VERIFY_PASSWORD_SUCCESS
-        );
+      .then(() =>
+      // Metrics.sendBehavioralMetric(
+      //   BEHAVIORAL_METRICS.VERIFY_PASSWORD_SUCCESS
+      // );
```
Comment (mkesavan13) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:1041]: I guess this code has been moved outside. I am also thinking this code should stay here and not outside. 

Kindly re-visit

**Review by mkesavan13 — CHANGES_REQUESTED** [review]
Please re-work on the commit types. 

- The test file changes shall be in a separate commit with type `test`
- The meeting info & meeting changes shall be under the type `fix`

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.js (hunk 1 above)
Comment (rajeshtezu90) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.js:167]: Then we will need to throw the caught error directly this way on line number 156. Error thrown from webex.request's catch throws error in this format and from here again we format it making message like `<message>, <code>, <code>` i.e. code gets repeated twice. I think we should format at line 167 and throw directly from 156 to cover `conversation.get` call as well.

Please confirm.

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js (hunk 2 above)
Comment (rajeshtezu90) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:971]: As per our discussion Metrics are added in `meetingInfoProvider.fetchMeetingInfo` which we are mocking in the unit test, as a result we will need to remove assertion for Metrics calls.

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js (hunk 3 above)
Comment (rajeshtezu90) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:1041]: Added it back.

**Comment by arun3528** [comment]
@rajeshtezu90  the unit test fix looks good. I did rerrange the test check to fix the error you can make the issue as resolved

**Pull request closed without merging**